lib/node_modules/@stdlib/array/base/cusome-by-right/docs/types/index.d.ts

```diff
@@ -0,0 +1,151 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { Collection, AccessorArrayLike } from '@stdlib/types/array';
+
+/**
+* Checks whether an element in a collection passes a test.
+*
+* @returns boolean indicating whether an element in a collection passes a test
+*/
+type Nullary<U> = ( this: U ) => boolean;
+
+/**
+* Checks whether an element in a collection passes a test.
+*
+* @param value - collection value
+* @returns boolean indicating whether an element in a collection passes a test
+*/
+type Unary<T, U> = ( this: U, value: T ) => boolean;
+
+/**
+* Checks whether an element in a collection passes a test.
+*
+* @param value - collection value
+* @param index - collection index
+* @returns boolean indicating whether an element in a collection passes a test
+*/
+type Binary<T, U> = ( this: U, value: T, index: number ) => boolean;
+
+/**
+* Checks whether an element in a collection passes a test.
+*
+* @param value - collection value
+* @param index - collection index
+* @param collection - input collection
+* @returns boolean indicating whether an element in a collection passes a test
+*/
+type Ternary<T, U> = ( this: U, value: T, index: number, collection: Collection<T> ) => boolean;
+
+/**
+* Checks whether an element in a collection passes a test.
+*
+* @param value - collection value
+* @param index - collection index
+* @param collection - input collection
+* @returns boolean indicating whether an element in a collection passes a test
+*/
+type Predicate<T, U> = Nullary<U> | Unary<T, U> | Binary<T, U> | Ternary<T, U>;
+
+/**
+* Interface describing `cusomeByRight`.
+*/
+interface CusomeByRight {
+	/**
+	* Cumulatively tests whether at least n array elements in a provided array pass a test implemented by a predicate function, while iterating from right-to-left.
+	*
+	* @param x - input array
+	* @param n - number of elements
+	* @param predicate - test function
+	* @param thisArg - execution context
+	* @returns output array
+	*
+	* @example
+	* function isPositive( v ) {
+	* 	return v > 0;
```

Comment:
```suggestion
	*     return v > 0;
```

No use of tabs in JSDoc comments.